src/net.rs

```diff
@@ -120,13 +148,24 @@ impl Gossip {
             }
             .instrument(error_span!("gossip", %me)),
         );
-        Self {
+
+        Ok(Gossip {
             to_actor_tx,
             _actor_handle: Arc::new(AbortOnDropHandle::new(actor_handle)),
             max_message_size,
             next_receiver_id: Default::default(),
             #[cfg(feature = "rpc")]
             rpc_handler: Default::default(),
+        })
+    }
+}
+
+impl Gossip {
+    /// Creates a default `Builder`, with the endpoint set.
+    pub fn builder(endpoint: Endpoint) -> Builder {
```

Comment:
In the blobs builder I only require the endpoint on build, not initially when creating the builder.

Is the endpoint something that we need during the build to e.g. come up with good defaults, or can we also pass it later during build?

Sometimes you have to do multi-stage builders, like we have with the router, but in general I prefer builders to be as passive as possible and all the active stuff happening in the actual fn build.

Reply:
I mirrored this api with the `router`, where you can only construct the builder with the `endpoint`, as I like to have the required args upfront in the builder, aka the exact opposite of preference that you have 😅

Reply:
for me personally having `build`/`spawn` take arguments is an antipattern 🤷

Reply:
I will quickly argue my case, but as I said it's not he end of the world...

```rust
Foo::builder()
  .set_x(1)
  .set_y(2) // up to here this thing is a pure data thing, you don't have to think about its lifetime, you can pass it as an argument to another builder etc.
  .spawn(endpoint).await? // everything active happens here, it is async anyway, so you can call async fns on the endpoint
```

```rust
Foo::builder(endpoint) // this thing is heavy from the start since it keeps an endpoint ref. If you need something like the default relay from the endpoint you need to make fn builder(endpoint) async and fallible. Which I think is werid.
  .set_x(1)
  .set_y(2)
  .spawn().await
```